src/nvim/mark_extended.c

```diff
@@ -251,6 +253,30 @@ ExtendedMark *extmark_from_pos(buf_T *buf,
   return NULL;
 }
 
+// Return the next avaliable id in a namespace
+uint64_t extmark_next_id_get(buf_T *buf, uint64_t ns)
+{
+  uint64_t last_seen = 0;
+
+  if (!buf->b_extmark_ns) {
+    return last_seen;
+  }
+  ExtmarkNs *ns_obj = pmap_get(uint64_t)(buf->b_extmark_ns, ns);
+  if (!ns_obj) {
+    return last_seen;
+  }
+
+  // Just adds +1 to the largest id found
+  uint64_t mark_id;
+  ExtendedMark *extmark;
+  map_foreach(ns_obj->map, mark_id, extmark, {
```

Comment:
Good idea, Isn't this `O(n)` though? 
10 elements - > 20 operations
20 elements -> 40 operations